apmconnector/logs.go

```diff
@@ -0,0 +1,52 @@
+package apmconnector
+
+import (
+	"fmt"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func BuildTransactions(td ptrace.Traces) plog.Logs {
+	logs := plog.NewLogs()
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		resourceLogs := logs.ResourceLogs().AppendEmpty()
+		rs := td.ResourceSpans().At(i)
+		rs.Resource().CopyTo(resourceLogs.Resource())
+		for j := 0; j < rs.ScopeSpans().Len(); j++ {
+			scopeSpan := rs.ScopeSpans().At(j)
+			scopeLog := resourceLogs.ScopeLogs().AppendEmpty()
+			for k := 0; k < scopeSpan.Spans().Len(); k++ {
+				span := scopeSpan.Spans().At(k)
+				if span.Kind() != ptrace.SpanKindServer {
+					continue
+				}
+				log := scopeLog.LogRecords().AppendEmpty()
+				buildTransaction(log, span)
+			}
+		}
+	}
+	return logs
+}
+
+func buildTransaction(lr plog.LogRecord, span ptrace.Span) {
+	lr.Attributes().PutStr("event.domain", "newrelic.otel_collector")
+	lr.Attributes().PutStr("event.type", "Transaction")
+
+	httpRoute, routePresent := span.Attributes().Get("http.route")
+	method, methodPresent := span.Attributes().Get("http.method")
+	if routePresent {
+		var transactionName string
+		if methodPresent {
+			transactionName = fmt.Sprintf("%s (%s)", httpRoute.AsString(), method.AsString())
+		} else {
+			transactionName = httpRoute.AsString()
+		}
+		lr.Attributes().PutStr("name", transactionName)
```

Comment:
What do you think about doing the metric and log generating in the same loop?  We're going to have a lot of shared code between the two.  At the very least, we should probably share a function that, given a span, returns the transaction name.

Reply:
yeah, that sounds a lot more reasonable that what I'm currently doing